src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -5521,14 +5534,12 @@ mono_profiler_fire_buffered_gc_event_roots (
 		for (uint64_t i = 0; i < count; i++) {
 			// GCRoots.Values[].ObjectID.
 			object_id = (uintptr_t)SGEN_POINTER_UNTAG_ALL (*objects);
-			memcpy (buffer, &object_id, sizeof (object_id));
-			buffer += sizeof (object_id);
+			ep_write_buffer_uintptr_t (&buffer, object_id);
 			objects++;
 
 			// GCRoots.Values[].AddressID.
 			address_id = (uintptr_t)*objects;
```

Comment:
Heads-up, there is a fix on this line in main, so maybe you should rebase to make sure that is correctly picked up. It should use address instead of objects.